composer/utils/ddp.py

```diff
@@ -31,25 +78,182 @@ def _get_distributed_config_var(env_var: str,
 
 
 def get_world_size() -> int:
+    """Returns the DDP world size
+
+    Returns:
+        int: The world size
+    """
     return _get_distributed_config_var(env_var="WORLD_SIZE",
                                        human_name="world size",
                                        default=1,
                                        fetch_fn_name="get_world_size")
 
 
 def get_global_rank() -> int:
+    """Returns the global rank of the current process, which is on `[0, WORLD_SIZE - 1]`
+
+    Returns:
+        int: The global rank
+    """
     return _get_distributed_config_var(env_var="RANK", human_name="global rank", default=0, fetch_fn_name="get_rank")
 
 
 def get_local_world_size() -> int:
+    """Returns the local world size, which is the number of processes for the current node.
+
+    Returns:
+        int: The local world size
+    """
     return _get_distributed_config_var(env_var="LOCAL_WORLD_SIZE", human_name="local world size", default=1)
 
 
 def get_local_rank() -> int:
+    """Returns the local rank for the current process, which is on `[0, LOCAL_WORLD_SIZE - 1]`
+
+    Returns:
+        int: The local world size
+    """
     local_rank = _get_distributed_config_var(env_var="LOCAL_RANK", human_name="local rank", default=0)
     assert local_rank == get_global_rank() % get_local_world_size(), "invariant violation"
     return local_rank
 
 
-def is_rank_zero() -> bool:
-    return get_global_rank() == 0
+def barrier() -> None:
+    if dist.is_available():
+        dist.barrier()
+    # If not on DDP, then do nothing
+
+
+def all_reduce(
+    tensor: torch.Tensor,
+    reduce_operation: str = "SUM",
+) -> None:
+    if dist.is_available():
+        reduce_op = getattr(dist.ReduceOp, reduce_operation.upper())
+        dist.all_reduce(tensor, op=reduce_op)
+    else:
+        raise NotImplementedError("Non-DDP versions of reduce operations are not yet implemented")
```

Comment:
This should be correct, right? Since BYOT is now able to hit this function, it's a little more important for us to support non-distributed training runs.

```suggestion
        return
```

Reply:
Ah yes, correct, assuming that the world size is 1, then we can just return. I updated the guards in the DDP functions to reflect this.